File: client-app/src/app/models/buchung.ts
import { Tag } from './tag';

export interface Buchung {
  id: string;
  name: string;
  beschreibung: string;
  zeitpunkt: Date | null;
  betrag: number | null;
  kategorie: number;
  intervall: number;
  created?: Date | null;
  updated?: Date | null;
  tags: Tag[];
}

export class Buchung implements Buchung {
  constructor(init?: BuchungFormValues) {
    Object.assign(this, init);
  }
}

export class BuchungFormValues {
  id?: string = undefined;
  name: string = '';
  beschreibung: string = '';
  zeitpunkt: Date | null = null;
  betrag: number | null = null;
  kategorie: number = 0;
  intervall: number = 0;
  created?: Date | null = null;
  updated?: Date | null = null;
  tags: Tag[] = [];

  constructor(buchung?: BuchungFormValues) {
    if (buchung) {
      this.id = buchung.id;
      this.name = buchung.name;
      this.beschreibung = buchung.name;
      this.kategorie = buchung.kategorie;
      this.intervall = buchung.intervall;
      this.zeitpunkt = buchung.zeitpunkt;
      this.betrag = buchung.betrag;
      this.tags = buchung.tags;
    }
  }
}

File: BackEnd.Net/Domain/Buchung.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    /// <summary>
    /// Repräsentiert eine Buchung
    /// </summary>
    public record Buchung
    {
        /// <summary>
        /// Technischer Schlüssel
        /// </summary>
        public Guid Id { get; init; }
        
        /// <summary>
        /// Der Name der Buchung
        /// </summary>
        public string Name { get; init; }
        
        /// <summary>
        /// Eine Beschreibung
        /// </summary>
        public string Beschreibung { get; init; }

        /// <summary>
        /// Der Zeitpunkt der Buchung, wann sie für gewöhnlich fällig ist
        /// </summary>
        public DateTime Zeitpunkt { get; init; }
        
        /// <summary>
        /// Der Betrag der Buchung. Hierbei handelt es sich um dem Monatliche
        /// Betrag.
        /// </summary>
        public decimal Betrag { get; init; }
        
        /// <summary>
        /// Die Kategorie, ob es ein Ein- oder Abgang ist.
        /// </summary>
        public Kategorie Kategorie { get; init; }
        
        /// <summary>
        /// Wie oft die Buchung ansteht. Aus Betrag und Intervall kann die
        /// eigentliche Summe errechnet werden.
        /// </summary>
        public Intervall Intervall { get; init; }
        
        /// <summary>
        /// Das Erstellungsdatum der Buchung
        /// </summary>
        public DateTime Created { get; init; }
        
        /// <summary>
        /// Die letzte Aktualisierung der Buchung
        /// </summary>
        public DateTime? Updated { get; init; }

        /// <summary>
        /// Zeigt an, ob die Buchung gelöscht ist
        /// </summary>
        public bool IsDeleted { get; init; }

        /// <summary>
        /// Gesetzte Tags
        /// </summary>
        public ICollection<Tag> Tags { get; init; } = new List<Tag>();
    }
}